src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+        /// </summary>
+        public ResolvedServicePartition ResolvedServicePartition { get => this.InnerClient.ResolvedServicePartition; set => this.InnerClient.ResolvedServicePartition = value; }
+
+        /// <summary>
+        /// Gets or Sets the name of the listener in the replica or instance to which the client is connected to.
+        /// </summary>
+        public string ListenerName { get => this.InnerClient.ListenerName; set => this.InnerClient.ListenerName = value; }
+
+        /// <summary>
+        /// Gets or Sets the service endpoint to which the client is connected to.
+        /// </summary>
+        public ResolvedServiceEndpoint Endpoint { get => this.InnerClient.Endpoint; set => this.InnerClient.Endpoint = value; }
+
+        /// <summary>
+        /// Gets the inner client which this client wraps.
+        /// </summary>
+        public IServiceRemotingClient InnerClient { get; private set; }
+
+        /// <summary>
+        /// Sends a message to the service and gets a response back. The correlation id and context are sent along with
+        /// the message as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        /// <returns>Response body.</returns>
+        public Task<byte[]> RequestResponseAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            return SendAndTrackRequestAsync(messageHeaders, requestBody, () => this.InnerClient.RequestResponseAsync(messageHeaders, requestBody));
+        }
+
+        /// <summary>
+        /// Sends a one-way message to the service. The correlation id and context are sent along with the message
+        /// as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        public void SendOneWay(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            SendAndTrackRequestAsync(messageHeaders, requestBody, () =>
+            {
+                this.InnerClient.SendOneWay(messageHeaders, requestBody);
+                return Task.FromResult<byte[]>(null);
+            }).Forget();
+        }
+
+        private async Task<byte[]> SendAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody, Func<Task<byte[]>> doSendRequest)
+        {
+            DependencyTelemetry dt = new DependencyTelemetry();
+            dt.Name = ServiceRemotingLoggingStrings.OutboundRequestActivityName;
+
+            // Determine if there is currently an activity. If there is, initializes the dependency telemetry with the activity information,
+            // which helps set the right context for the dependency telemetry
+            Activity currentActivity = Activity.Current;
+            if (currentActivity != null)
```

Comment:
this does not look right
It should be similar to 
```
public async Task Enqueue(string payload)
{
    // StartOperation is a helper method that initializes the telemetry item
    // and allows correlation of this operation with its parent and children.
    var operation = telemetryClient.StartOperation<DependencyTelemetry>("enqueue " + queueName);
    operation.Telemetry.Type = "Queue";
    operation.Telemetry.Data = "Enqueue " + queueName;

    ...
    try
    {
         ...
    }
    finally
    {
        telemetryClient.StopOperation(operation);
    }
}
```
I.e. StartOperation should be enough.

1. Ids should be set to the new Activity, not the parent activity.
2. Ids should be set by AI OperationCorrelaitonTelemetryIntiializer unless there are strong reasons not to use it
3. StartOperation initializes telemetry and assigns all the properties: there should be no reason to call fabricTelemetryInitializer.Initialize(dt);